### PR TITLE
fix: use a rounded corner decoration by default for ShadTabs to match the official shadcn implementation

### PR DESCRIPTION
**Pull request by helightdev**
Comparing https://ui.shadcn.com/docs/components/tabs with https://flutter-shadcn-ui.mariuti.com/components/tabs/ shows the difference and I think it would be beneficial to adopt the original design because the sharp corners stand out quite a bit.

### CODE REVIEW

**Comment by helightdev** [comment]
Just noticed that I should probably also change ShadDefaultThemeNoSecondaryBorderVariant

**Comment by nank1ro** [comment]
Please enable the dart formatter otherwise I can't figure out what lines you changed.
For example in VScode you can setup it to run on file save.

Yes the no-secondary-border-variant must be adapted too

**Comment by helightdev** [comment]
I did format it, but seemingly my ide formatter somehow did it differently than the command line formatter, I'll do that after changing the other file in the final commit.

**Comment by helightdev** [comment]
Alright should be done now.

File: lib/src/theme/themes/default_theme_no_secondary_border_variant.dart [hunk 1]
```diff
@@ -756,6 +756,9 @@ class ShadDefaultThemeNoSecondaryBorderVariant extends ShadThemeVariant {
       tabForegroundColor: colorScheme.foreground,
       tabSelectedForegroundColor: colorScheme.foreground,
       tabSelectedShadows: ShadShadows.sm,
+      tabDecoration: ShadDecoration(
+        border: ShadBorder.all(radius: radius, width: 0),
```
Comment (nank1ro) [inline lib/src/theme/themes/default_theme_no_secondary_border_variant.dart:760]: Update the radius to 4
Screenshot 2024-12-02 at 10 41 38

As you can see in the image above, the default radius used by ui.shadcn.com is 4px.
0.5 rem = 8 px
8px - 4px = 4px

File: lib/src/theme/themes/default_theme_variant.dart [hunk 2]
```diff
@@ -728,6 +728,9 @@ class ShadDefaultThemeVariant extends ShadThemeVariant {
       tabForegroundColor: colorScheme.foreground,
       tabSelectedForegroundColor: colorScheme.foreground,
       tabSelectedShadows: ShadShadows.sm,
+      tabDecoration: ShadDecoration(
+        border: ShadBorder.all(radius: radius, width: 0),
```
Comment (nank1ro) [inline lib/src/theme/themes/default_theme_variant.dart:732]: Update the radius to 4 as mentioned above

File: lib/src/theme/themes/default_theme_no_secondary_border_variant.dart (hunk 1 above)
Comment (nank1ro) [reply lib/src/theme/themes/default_theme_no_secondary_border_variant.dart:760]: Just checked the source code and it uses `rounded-sm` which is 2px.
https://github.com/shadcn-ui/ui/blob/805ed4120a6a8ae6f6e9714cbd776e18eeba92c7/apps/www/registry/default/ui/tabs.tsx#L32C66-L32C76

So the source code and what is rendered on the docs is different.
No idea of what is supposed to be the right value.
Probably 4px is more elegant than just 2px

**Comment by nank1ro** [comment]
implemented in 1595a16521c022015a4a22d161d73ca4908e6197

**Pull request closed without merging**